### PR TITLE
Fixed bugs in explorer redirection

### PR DESCRIPTION
**Pull request by paulstn**
### Description
- allow for redirection in associated objects skipping indices
- redirection in datasources table implemented
- extended associated object type
- fixed url redirection to stop causing crashes and unexpected issues, used mainly in the discover to explorer flow
- extended the query text area character limit now to 2^14
- change the direct query visualization error to open acc flyout in Q.W.

- s3 no results page changes with sample queries and glue data cache support to populate a functioning query, shown in the following flows:|

s3 PPL empty page (no query run yet):
image
s3 SQL empty page (no query run yet):
image
s3 no results (post query run w/ no data returned):
image

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/types/data_connections.ts [hunk 1]
```diff
@@ -35,11 +35,12 @@ export interface Acceleration {
 }
 
 export interface AssociatedObject {
+  tableName: string;
```
Comment (paulstn) [inline common/types/data_connections.ts:38]: @RyanL1997 @sejli are you fine with this additional field in associated object? using it for defining the table name in skipping index redirection

Comment (sejli) [reply common/types/data_connections.ts:38]: Sounds good to me, as long as it's updated wherever this type is used.

**Review by ps48 — APPROVED** [review]

File: common/constants/autocomplete.ts [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 import { BaseItem } from '@algolia/autocomplete-core';
 
+export const textAreaMaxLength = 16384;
```
Comment (Swiddis) [inline common/constants/autocomplete.ts:9]: Just curious: where is this limit from? I don't see 2^14 often.

Comment (paulstn) [reply common/constants/autocomplete.ts:9]: arbitrary amount where I looked at how long certain sql queries could be, i made a value judgement. we can always change it later, though

File: public/components/event_analytics/explorer/explorer.tsx [hunk 3]
```diff
@@ -276,28 +276,32 @@ export const Explorer = ({
...
-      );
-      dispatch(
-        changeQuery({
-          tabId,
-          query: { [RAW_QUERY]: queryToRun },
-        })
-      );
-      setTempQuery(queryToRun);
+      if (datasourceName && datasourceType) {
+        dispatch(
```
Comment (Swiddis) [inline public/components/event_analytics/explorer/explorer.tsx:280]: Another question out of curiosity: what's the benefit of `dispatch` over `useEffect`? I'm not sure what'd motivate me to use one over the other, I haven't looked deeply into redux.

**Review by Swiddis — APPROVED** [review]

File: public/components/event_analytics/explorer/explorer.tsx (hunk 3 above)
Comment (paulstn) [reply public/components/event_analytics/explorer/explorer.tsx:280]: the extent of my knowledge is that dispatch would be used for updating redux state, i'm not sure about if useEffect could be used for the same thing

**Pull request merged**